### PR TITLE
[luau] Add new port

### LINKED ISSUE #21210: [New Port Request] Luau (Lua with types)
Library name: Luau

Library description: 
An extension to Lua, that adds types, and many more features that are listed on the Luau website.

Source repository URL: 
https://github.com/Roblox/luau

Project homepage (if different from the source repository): 
https://luau-lang.org/

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by raxracks**
Fixes #21210 (already closed, but a PR was never made)

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by raxracks** [comment]
@microsoft-github-policy-service agree

File: ports/luau/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Roblox/luau
+    REF 0.594
```
Comment (MonicaLiu0311) [inline ports/luau/portfile.cmake:4]:
```suggestion
    REF "${VERSION}"

```

File: ports/luau/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Roblox/luau
+    REF 0.594
+    SHA512 883e18ef93784f5625339b6b37c02cd0dc5dbc9063c4b622380b660b5e0df41f89c6062f64ef44c1f7b0e8f4a275552e363a50cedebea2bbe14943de29fb5e5c
+    HEAD_REF master
+    PATCHES
+        cmake-install.patch
+)
+
+vcpkg_cmake_configure(SOURCE_PATH ${SOURCE_PATH})
```
Comment (MonicaLiu0311) [inline ports/luau/portfile.cmake:11]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH ${SOURCE_PATH}
    OPTIONS
        -DLUAU_BUILD_TESTS=OFF
)
```
Don't build test related.

**Comment by raxracks** [comment]
Thanks, I've resolved them

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/luau/cmake-install.patch [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+index 15a1b8a..9ed385c 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -92,6 +96,22 @@ target_include_directories(isocline PUBLIC extern/isocline/include)
+ 
+ target_include_directories(Luau.VM.Internals INTERFACE VM/src)
+ 
++install(
++    TARGETS Luau.VM Luau.CodeGen Luau.Common Luau.Ast Luau.Analysis Luau.Compiler
++    EXPORT LuauTargets
```
Comment (dg0yt) [inline ports/luau/cmake-install.patch:11]: AFAICS this export is unused.

File: ports/luau/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS
+        -DLUAU_BUILD_TESTS=OFF
+)
+vcpkg_cmake_install()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
```
Comment (dg0yt) [inline ports/luau/portfile.cmake:21]: What is removed here? Any tools which should be copied?

File: ports/luau/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  "homepage": "https://luau-lang.org",
+  "license": "MIT",
+  "supports": "!x86 & !uwp & !arm32",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
```
Comment (dg0yt) [inline ports/luau/vcpkg.json:14]: AFAICS this depedency is unused.

File: ports/luau/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    REF "${VERSION}"
+
+    SHA512 883e18ef93784f5625339b6b37c02cd0dc5dbc9063c4b622380b660b5e0df41f89c6062f64ef44c1f7b0e8f4a275552e363a50cedebea2bbe14943de29fb5e5c
+    HEAD_REF master
+    PATCHES
+        cmake-install.patch
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (dg0yt) [inline ports/luau/portfile.cmake:13]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"

```

**Review by BillyONeal — CHANGES_REQUESTED** [review]
(I agree with @dg0yt's comments)

**Comment by MonicaLiu0311** [comment]
Ping @raxracks

**Comment by MonicaLiu0311** [comment]
Ping @raxracks again

Is there any new progress?

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**